### PR TITLE
AIP-84: Migrating DELETE queued asset events for assets to FASTAPI

### PR DESCRIPTION
**Pull request by vatsrahul1001**
Related: https://github.com/apache/airflow/issues/42370

Swagger Specs
image
image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pierrejeambrun — APPROVED** [review]
Nice, ready to merge, restarting failed checks

File: tests/api_fastapi/core_api/routes/public/test_assets.py [hunk 1]
```diff
@@ -549,3 +549,30 @@ def test_invalid_attr_not_allowed(self, test_client, session):
...
+    def test_should_respond_204(self, test_client, session, create_dummy_dag):
+        dag, _ = create_dummy_dag()
+        dag_id = dag.dag_id
+        uri = "s3://bucket/key/1"
+        self.create_assets(session=session, num=1)
+        asset_id = 1
+        self._create_asset_dag_run_queues(dag_id, asset_id, session)
+
+        response = test_client.delete(
+            f"/public/assets/queuedEvent/{uri}",
```
Comment (pierrejeambrun) [inline tests/api_fastapi/core_api/routes/public/test_assets.py:787]: Maybe we should take the opportunity to `assets/queudEvents` to be consistent with other routes, especially `assets/events`

**Comment by vatsrahul1001** [comment]
> Won't pass, tests need fixing

taking a look

**Review by pierrejeambrun — APPROVED** [review]

File: tests/api_fastapi/core_api/routes/public/test_assets.py (hunk 1 above)
Comment (pierrejeambrun) [reply tests/api_fastapi/core_api/routes/public/test_assets.py:787]: Follow up PR, resolving

**Pull request merged**